pkg/datapath/linux/ipsec.go

```diff
@@ -320,7 +320,7 @@ func (n *linuxNodeHandler) enableIPSecIPv4Do(newNode *nodeTypes.Node, nodeID uin
 		statesUpdated = false
 	}
 
-	if n.datapathConfig.TunnelDevice == "" {
+	if n.nodeConfig.RoutingMode != option.RoutingModeTunnel {
```

Comment:
What about `option.Config.TunnelingEnabled()`? We already use it in this file.

Reply:
Was thinking this, and actually waiting for someone in foundations to chime in. But IIRC, we want to move away from reaching into the global daemon config, since at some point, it will be gone. Didn't want to further add to references to it.